diff --git a/ubiquity/plugins/ubi-language.py b/ubiquity/plugins/ubi-language.py
index 90a1891..05c81e8 100644
--- a/ubiquity/plugins/ubi-language.py
+++ b/ubiquity/plugins/ubi-language.py
@@ -504,8 +504,8 @@ class PageKde(PageBase):
         misc.drop_privileges()
         misc.drop_privileges_save()
         # copy over gtkrc-2.0 to get the themeing right
-        if os.path.exists("/usr/share/kubuntu-default-settings"):
-            shutil.copy("/usr/share/kubuntu-default-settings/" +
+        if os.path.exists("/usr/share/netrunner-default-settings"):
+            shutil.copy("/usr/share/netrunner-default-settings/" +
                         "dot-gtkrc-2.0-kde4",
                         os.getenv("HOME") + "/.gtkrc-2.0")
         QDesktopServices.openUrl(QUrl(url))
@@ -556,7 +556,7 @@ class PageKde(PageBase):
                 text = widget.text()
                 text = text.replace('${RELEASE}', release.name)
                 text = text.replace('${MEDIUM}', install_medium)
-                text = text.replace('Ubuntu', 'Kubuntu')
+                text = text.replace('Ubuntu', 'Netrunner')
                 widget.setText(text)
 
         self.update_release_notes_label()
diff --git a/ubiquity/plugins/ubi-prepare.py b/ubiquity/plugins/ubi-prepare.py
index 86a8ab5..368ed67 100644
--- a/ubiquity/plugins/ubi-prepare.py
+++ b/ubiquity/plugins/ubi-prepare.py
@@ -214,7 +214,7 @@ class PageKde(PreparePageBase):
         for widget in widgets:
             text = widget.text()
             text = text.replace('${RELEASE}', release.name)
-            text = text.replace('Ubuntu', 'Kubuntu')
+            text = text.replace('Ubuntu', 'Netrunner')
             widget.setText(text)
 
 
