diff --git a/scripts/plugininstall.py b/scripts/plugininstall.py
index 9272ead..74fec7a 100755
--- a/scripts/plugininstall.py
+++ b/scripts/plugininstall.py
@@ -199,7 +199,7 @@ class Install(install_misc.InstallBase):
 
         self.next_region()
         self.db.progress('INFO', 'ubiquity/install/apt')
-        self.configure_apt()
+        #self.configure_apt()
 
         self.configure_plugins()
 
@@ -231,10 +231,10 @@ class Install(install_misc.InstallBase):
         self.next_region()
         self.db.progress('INFO', 'ubiquity/install/installing')
 
-        if 'UBIQUITY_OEM_USER_CONFIG' in os.environ:
-            self.install_oem_extras()
-        else:
-            self.install_extras()
+        #if 'UBIQUITY_OEM_USER_CONFIG' in os.environ:
+            #self.install_oem_extras()
+        #else:
+            #self.install_extras()
 
         self.next_region()
         self.db.progress('INFO', 'ubiquity/install/bootloader')
@@ -465,7 +465,7 @@ class Install(install_misc.InstallBase):
         except debconf.DebconfError:
             domain = ''
         if hostname == '':
-            hostname = 'ubuntu'
+            hostname = 'netrunner'
 
         with open(self.target_file('etc/hosts'), 'w') as hosts:
             print("127.0.0.1\tlocalhost", file=hosts)
