diff --git a/gui/qt/app.ui b/gui/qt/app.ui
index 03d715c..0e2e254 100644
--- a/gui/qt/app.ui
+++ b/gui/qt/app.ui
@@ -148,7 +148,7 @@
           <item>
            <widget class="QLabel" name="distro_name_label">
             <property name="text">
-             <string>Kubuntu</string>
+             <string>Netrunner</string>
             </property>
             <property name="scaledContents">
              <bool>false</bool>
diff --git a/gui/qt/stepPartAuto.ui b/gui/qt/stepPartAuto.ui
index 01d6d4c..79a6caf 100644
--- a/gui/qt/stepPartAuto.ui
+++ b/gui/qt/stepPartAuto.ui
@@ -31,7 +31,7 @@
    <item>
     <widget class="QLabel" name="part_auto_comment_label">
      <property name="text">
-      <string>Where would you like to install Kubuntu?</string>
+      <string>Where would you like to install Netrunner?</string>
      </property>
     </widget>
    </item> 
