diff --git a/plasma/contents/code/LabelSlider.py b/plasma/contents/code/LabelSlider.py
index a2f13e3..6d87858 100644
--- a/plasma/contents/code/LabelSlider.py
+++ b/plasma/contents/code/LabelSlider.py
@@ -104,7 +104,7 @@ class LabelSlider(Plasma.Slider):
         self.label.setMaximumWidth(w)
         # set abs position of label, respect fonts-size
         h = self.size().height()
-        s = Plasma.Theme.defaultTheme().fontMetrics().height() + 2
+        s = Plasma.Theme.defaultTheme().fontMetrics().height() + 10
         v = int ((h/2) - s)
         self.label.setPos(0,v)
         