--- /home/leszek/Downloads/work/netrunner/13.12/iso/edit/usr/share/initramfs-tools/scripts/casper	2013-08-02 11:01:39.000000000 +0200
+++ /home/leszek/Downloads/work/netrunner/13.12/iso/64bit/netrunner1312-64/edit/usr/share/initramfs-tools/scripts/casper	2013-12-16 11:16:50.237058897 +0100
@@ -446,7 +446,14 @@
     mkdir -p /cow
     cowdevice="tmpfs"
     cow_fstype="tmpfs"
-    cow_mountopt="rw,noatime,mode=755"
+    #cow_mountopt="rw,noatime,mode=755"
+    # Low memory RAM create a bigger ram drive for ubiquity to succeed with 1 GB RAM only
+    TotalMem=$(cat /proc/meminfo | grep -i memtotal | cut -d ':' -f 2 | awk '{print $1}')
+    if [ "$TotalMem" -gt "800000" ] && [ "$TotalMem" -le "1500000" ]; then
+	cow_mountopt="rw,noatime,mode=755,size=600M"
+    else 
+	cow_mountopt="rw,noatime,mode=755"
+    fi
 
     # Looking for "${root_persistence}" device or file
     if [ -n "${PERSISTENT}" ]; then
