Description: Make it build against qt4
Author: Leszek Lesner <leszek@zevenos.com>

--- transmission-2.82.orig/qt/file-tree.cc
+++ transmission-2.82/qt/file-tree.cc
@@ -884,8 +884,9 @@ FileTreeView :: FileTreeView (QWidget *
 
   for (int i=0; i<NUM_COLUMNS; ++i)
     {
-      setColumnHidden (i, (i<FIRST_VISIBLE_COLUMN) || (LAST_VISIBLE_COLUMN<i));
-      header()->setSectionResizeMode(i, QHeaderView::Interactive);
+      /*setColumnHidden (i, (i<FIRST_VISIBLE_COLUMN) || (LAST_VISIBLE_COLUMN<i));
+      header()->setSectionResizeMode(i, QHeaderView::Interactive); */
+      header()->setResizeMode( i, QHeaderView::Interactive );
     }
 
   connect (this, SIGNAL(clicked(const QModelIndex&)),
--- transmission-2.82.orig/qt/favicon.cc
+++ transmission-2.82/qt/favicon.cc
@@ -14,7 +14,8 @@
 #include <QNetworkAccessManager>
 #include <QNetworkReply>
 #include <QNetworkRequest>
-#include <QStandardPaths>
+//#include <QStandardPaths> // No we don't want Qt5
+#include <QDesktopServices>
 
 #include "favicon.h"
 
@@ -40,7 +41,11 @@ Favicons :: ~Favicons( )
 QString
 Favicons :: getCacheDir( )
 {
+    /* No Qt5 please
     const QString base = QStandardPaths::writableLocation (QStandardPaths::CacheLocation);
+    return QDir( base ).absoluteFilePath( "favicons" ); */
+
+    const QString base = QDesktopServices::storageLocation( QDesktopServices::CacheLocation );
     return QDir( base ).absoluteFilePath( "favicons" );
 }
 
